File: GuiaApe06/Ejercicio6.java
import java.util.ArrayList; // Librería para usar ArrayList
import java.util.Scanner;   // Librería para entrada de datos

// Clase Producto
class Producto {

    // Atributos del producto
    String nombre;
    int cantidad;
    double precio;

    // Constructor para inicializar atributos
    public Producto(String nombre, int cantidad, double precio) {

        this.nombre = nombre;
        this.cantidad = cantidad;
        this.precio = precio;
    }

    // Método para mostrar información del producto
    public void mostrar() {

        System.out.println("Nombre: " + nombre);
        System.out.println("Cantidad: " + cantidad);
        System.out.println("Precio: $" + precio);
        System.out.println("-------------------");
    }
}

public class Ejercicio6{

    public static void main(String[] args) {

        // Crear objeto Scanner para leer datos
        Scanner sc = new Scanner(System.in);

        // Crear ArrayList para almacenar productos
        ArrayList<Producto> inventario = new ArrayList<>();

        // Variable para cantidad de productos
        int n;

        // Pedir cantidad de productos
        System.out.print("Cantidad de productos: ");
        n = sc.nextInt();

        // Limpiar buffer
        sc.nextLine();

        // Ciclo para registrar productos
        for(int i = 0; i < n; i++) {

            // Mostrar número del producto
            System.out.println("\nProducto " + (i + 1));

            // Pedir nombre del producto
            System.out.print("Nombre: ");
            String nombre = sc.nextLine();

            // Pedir cantidad
            System.out.print("Cantidad: ");
            int cantidad = sc.nextInt();

            // Pedir precio
            System.out.print("Precio: ");
            double precio = sc.nextDouble();

            // Limpiar buffer
            sc.nextLine();

            // Agregar producto al ArrayList
            inventario.add(new Producto(nombre, cantidad, precio));
        }

        // Mostrar título
        System.out.println("\n===== INVENTARIO =====");

        // Recorrer y mostrar productos
        for(Producto p : inventario) {

            // Mostrar información del producto
            p.mostrar();
        }

        // Cerrar Scanner
        sc.close();
    }
}

File: GuiaApe06/Ejercicio6.cpp
#include <iostream>   // Librería para entrada y salida de datos
#include <vector>     // Librería para usar vectores dinámicos

using namespace std;

// Clase Producto
class Producto {

public:

    // Atributos del producto
    string nombre;
    int cantidad;
    float precio;

    // Constructor para inicializar atributos
    Producto(string n, int c, float p) {

        nombre = n;
        cantidad = c;
        precio = p;
    }

    // Método para mostrar información del producto
    void mostrar() {

        cout << "Nombre: " << nombre << endl;
        cout << "Cantidad: " << cantidad << endl;
        cout << "Precio: $" << precio << endl;
        cout << "-------------------" << endl;
    }
};

int main() {

    // Vector para almacenar productos
    vector<Producto> inventario;

    // Variable para cantidad de productos
    int n;

    // Pedir cantidad de productos
    cout << "Cantidad de productos: ";
    cin >> n;

    // Ciclo para registrar productos
    for(int i = 0; i < n; i++) {

        // Variables auxiliares
        string nombre;
        int cantidad;
        float precio;

        // Mostrar número de producto
        cout << "\nProducto " << i + 1 << endl;

        // Limpiar buffer
        cin.ignore();

        // Pedir nombre del producto
        cout << "Nombre: ";

        // Leer nombre completo
        getline(cin, nombre);

        // Pedir cantidad
        cout << "Cantidad: ";
        cin >> cantidad;

        // Pedir precio
        cout << "Precio: ";
        cin >> precio;

        // Agregar producto al vector
        inventario.push_back(Producto(nombre, cantidad, precio));
    }

    // Mostrar título
    cout << "\n===== INVENTARIO =====" << endl;

    // Recorrer vector y mostrar productos
    for(auto p : inventario) {

        // Mostrar datos del producto
        p.mostrar();
    }

    return 0;
}
